routes/usersRouter.js

```diff
@@ -0,0 +1,112 @@
+const express = require("express");
+const router = express.Router();
+const Users = require("../models/schemas/users");
+const bcrypt = require('bcrypt');
+
+router.get("/", (req, res) => {
+  Users.find({}, (err, users) => {
+    if (err) {
+      console.log(err);
+      res.status(500).json({ status: 'Not OK', err });
+    } else {
+      res.status(200).json({ status: 'OK', users });
+    }
+  });
+});
+
+router.post("/signup", async (req, res) => {
+  if (req.body.password !== req.body.passwordCheck) {
+    return res.status(400).json({ status: 'Not OK', err: 'Passwords Don\'t match' })
+  }
+  req.body.password = await bcrypt.hash(req.body.password, 10)
+  new Users(req.body).save((err, result) => {
+    if (err) {
+      console.log(err);
+      res.status(500).json({ status: 'Not OK', err });
+    } else {
+      res.status(200).send({ status: 'OK', result });
+    }
+  });
+});
+
+router.post("/search", (req, res) => {
+  const query = {};
+  if (req.body.username) {
+    query.username = req.body.username;
+  }
+  if (req.body.emailAddress) {
+    query.emailAddress = req.body.emailAddress;
+  }
+  if (req.body.password) {
+    query.password = req.body.password;
+  }
```

Comment:
You probably don't want to search for users via password, bit of a security hole!

Reply:
Yeah, good point, hadn't noticed that